server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailGetMethod.scala

```diff
@@ -101,11 +102,16 @@ class EmailGetMethod @Inject() (serializer: Serializer,
     request.properties match {
       case None => Right(Email.defaultProperties)
       case Some(properties) =>
-        val invalidProperties = properties -- Email.allowedProperties
-        if (invalidProperties.isEmpty()) {
+        val mayContainsSpecificHeaders = properties -- Email.allowedProperties
+        if (mayContainsSpecificHeaders.isEmpty()) {
           Right(properties ++ Email.idProperty)
         } else {
-          Left(new IllegalArgumentException(s"The following properties [${invalidProperties.format()}] do not exist."))
+          val invalidProperties = mayContainsSpecificHeaders.value.filter(p => !p.value.startsWith(SPECIFIC_HEADER_PREFIX))
+          if (invalidProperties.nonEmpty) {
+            Left(new IllegalArgumentException(s"The following properties [${Properties(invalidProperties).format()}] do not exist."))
```

Comment:
Maybe we would benefit from having some object encapsulation. Let me propose:

```
object SpecificHeaderProperty {
   def of(property: Property): Option[SpecificHeaderProperty] = ???
}

case class SpecificHeaderProperty(headerName: value) {
    def retrieve(mime4JMessage: Message): SpecificHeader = ???
}

case class SpecificHeader(property: Property, value: Option[EmailHeaderValue])
```

The parsing rule will be better encapsulated, less logic will be leaking. We could be looking for each client supplied property if it is either a statically specified allowed property, or can be interpreted as a specific header.

Maybe a :

```
sealed trait EmailPropertyValidation
case class Unknown(property: Property)
case class SpecificHeaderProperty ...
case class AllowedProperty(property: Property)
```

Helpfull too.